### PR TITLE
Add benchmarks for seviri hrit

### PR DESCRIPTION
**Pull request by mraspaud**
This PR adds download/demo methods for SEVIRI data in HRIT format and some benchmarks

 - [x] Tests added 
 - [ ] Fully documented

### CODE REVIEW

File: satpy/demo/seviri_hrit.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    if subset is None:
+        subset = _create_full_set()
+    pattern = os.path.join(base_dir, FILENAME)
+    files = []
+    for key, value in subset.items():
+        if key in ["PRO", "EPI"]:
+            files.append(pattern.format(channel="", segment=key))
+        else:
+            for segment in value:
+                files.append(pattern.format(channel=key, segment=f"{segment:06d}"))
```
Comment (djhoese) [inline satpy/demo/seviri_hrit.py:67]: You're more "clean code"-y than I am, but this seems like a lot of levels of indentation.

File: benchmarks/ahi_hsd_benchmarks.py [hunk 2]
```diff
@@ -74,18 +73,10 @@ def peakmem_save_true_color_to_geotiff(self):
         """Check peak memory usage of the generation and saving of true_color_nocorr."""
         self.save_true_color_nocorr_as_geotiff()
 
-    def load(self, composite):
-        """Load one composite."""
-        from satpy import Scene
-        scn = Scene(filenames=self.data_files, reader='ahi_hsd')
-        scn.load([composite], pad_data=False)
-        return scn
```
Comment (djhoese) [inline benchmarks/ahi_hsd_benchmarks.py:82]: I could be missing something but how can you remove this? All the other `compute_X` methods use `.load` still.

**Review by djhoese — COMMENTED** [review]
One clean code suggestion, one confusion with possibly broken code, but otherwise I'm guessing this filesystem stuff in the tests avoids touching disk completely? Is this something the VIIRS SDR tests could be updated with?

File: benchmarks/ahi_hsd_benchmarks.py (hunk 2 above)
Comment (mraspaud) [reply benchmarks/ahi_hsd_benchmarks.py:82]: This is now in the parent class

Comment (djhoese) [reply benchmarks/ahi_hsd_benchmarks.py:82]: But under a different name.

Comment (djhoese) [reply benchmarks/ahi_hsd_benchmarks.py:82]: `.load_no_padding` but `compute_B01` below still uses `.load`

Comment (mraspaud) [reply benchmarks/ahi_hsd_benchmarks.py:82]: doh, indeed, thanks for spotting this!

Comment (djhoese) [reply benchmarks/ahi_hsd_benchmarks.py:82]: Code review!

File: satpy/demo/seviri_hrit.py (hunk 1 above)
Comment (mraspaud) [reply satpy/demo/seviri_hrit.py]: Fixed

File: benchmarks/ahi_hsd_benchmarks.py (hunk 2 above)
Comment (mraspaud) [reply benchmarks/ahi_hsd_benchmarks.py:82]: Fixed

**Comment by mraspaud** [comment]
I'm running the benchmarks now locally, but I think this is ready

**Comment by mraspaud** [comment]
As for the filesystem stuff, maybe it could work for viirs, I can give it a try.

**Comment by mraspaud** [comment]
I'm merging this to have time to run all the benchmarks today on EWC

**Pull request merged**